packages/eslint-plugin/src/rules/consistent-indexed-object-style.ts

```diff
@@ -142,6 +143,69 @@ export default createRule<Options, MessageIds>({
             !node.extends.length,
           );
         },
+        TSMappedType(node): void {
+          const key = node.key;
+          const scope = context.sourceCode.getScope(key);
+
+          const scopeManagerKey = nullThrows(
+            scope.variables.find(
+              value => value.name === key.name && value.isTypeVariable,
+            ),
+            'key type parameter must be a defined type variable in its scope',
+          );
+
+          // If the key is used to compute the value, we can't convert to a Record.
+          if (
+            scopeManagerKey.references.some(
+              reference => reference.isTypeReference,
+            )
+          ) {
+            return;
+          }
+
+          const constraint = node.constraint;
+
+          if (
+            constraint.type === AST_NODE_TYPES.TSTypeOperator &&
+            constraint.operator === 'keyof' &&
+            !isParenthesized(constraint, context.sourceCode)
+          ) {
+            // This is a weird special case, since modifiers are preserved by
+            // the mapped type, but not by the Record type. So this type is not,
+            // in general, equivalent to a Record type.
+            return;
+          }
+
+          // There's no builtin Mutable<T> type, so we can't autofix it really.
+          const canFix = node.readonly !== '-';
+
+          context.report({
+            node,
+            messageId: 'preferRecord',
+            ...(canFix && {
+              fix: (fixer): ReturnType<ReportFixFunction> => {
+                const keyType = context.sourceCode.getText(constraint);
+                const valueType = context.sourceCode.getText(
+                  node.typeAnnotation,
+                );
+
+                let recordText = `Record<${keyType}, ${valueType}>`;
+
+                if (node.optional === '+' || node.optional === true) {
+                  recordText = `Partial<${recordText}>`;
+                } else if (node.optional === '-') {
+                  recordText = `Required<${recordText}>`;
```

Comment:
I'm realizing that `Required<Record<K, V>>` is probably always redundant, and I think can always just be `Record<K, V>`

Reply:
Talked 1:1 - tricky and maybe not determinable. Sticking with it as-is.